interfaces/Config/templates/config_server.tmpl

```diff
@@ -223,6 +234,19 @@
                             <label class="config" for="displayname$cur">$T('srv-displayname')</label>
                             <input type="text" name="displayname" id="displayname$cur" value="$server['displayname']" />
                         </div>
+                        <div class="field-pair advanced-settings">
+                            <label class="config" for="expire_date">$T('srv-expire_date')</label>
+                            <input type="date" name="expire_date" id="expire_date$cur"  value="$server['expire_date']" />
+                            <span class="desc">$T('srv-explain-expire_date')</span>
+                        </div>
+                        <div class="field-pair advanced-settings">
+                            <label class="config" for="quota_left">$T('srv-quota_left')</label>
+                            <input type="text" name="quota_left" id="quota_left$cur" value="$server['quota_left']" />
+                            <input type="hidden" name="quota_left_old" id="quota_left_old$cur" value="$server['quota_left']" />
+                            <input type="hidden" name="usage_at_start" id="usage_at_start$cur" value="$server['usage_at_start']" />
+                            <input type="hidden" name="usage_at_start_new" id="usage_at_start_new$cur"  value="<!--#if 'amounts' in $server#-->$(server['amounts'][0])<!--#end if#-->" />
```

Comment:
I checked out the code and still don't understand why there need to be these 3 fields?

Reply:
Or should these be hidden?

Reply:
The warning should be sent if `quota_left` (user input) + `usage_at_start` (what the BPSMeter grand_total is when the user enters quota_left) > `grand_total` (whenever the check is run).

If the user does a change to the server but the `quota_left` is not changed, the old `usage_at_start` should be kept. To know that the user has not changed quota_left, we need to know what it was before the save (unless there's another way to detect it). Ideally that could be done by loading the old server data before the new ones are saved, but I couldn't figure out how to do that. That's why I store it in the form as `quota_left_old`.

If the user has changed `quota_left`, we need replace usage_at_start with new data from BPSMeter grand_total. I couldn't figure out how to get that variable either in set_dict, so it's sent from the form as usage_at_start_new.

Yes, all three fields should be hidden.